exp/runtime/internal/controllers/extensionconfig_controller.go

```diff
@@ -180,19 +184,19 @@ func (r *Reconciler) secretToExtensionConfig(secret client.Object) []reconcile.R
 	result := []ctrl.Request{}
 
 	extensionConfigs := runtimev1.ExtensionConfigList{}
-	if err := r.Client.List(context.Background(), &extensionConfigs); err != nil {
+	indexKey := secret.GetNamespace() + "/" + secret.GetName()
+
+	if err := r.Client.List(
+		context.Background(),
+		&extensionConfigs,
+		client.MatchingFields{injectCAFromSecretAnnotationField: indexKey},
+	); err != nil {
 		return nil
 	}
 
 	for _, ext := range extensionConfigs.Items {
-		if secretNameRaw, ok := ext.GetAnnotations()[runtimev1.InjectCAFromSecretAnnotation]; ok {
-			secretName := splitNamespacedName(secretNameRaw)
-			// append all extensions to the result which refer the object as secret
-			if secretName.Namespace == secret.GetNamespace() && secretName.Name == secret.GetName() {
-				name := client.ObjectKey{Namespace: ext.GetNamespace(), Name: ext.GetName()}
-				result = append(result, ctrl.Request{NamespacedName: name})
-			}
-		}
+		name := client.ObjectKey{Namespace: ext.GetNamespace(), Name: ext.GetName()}
+		result = append(result, ctrl.Request{NamespacedName: name})
```

Comment:
```suggestion
		result = append(result, ctrl.Request{NamespacedName: client.ObjectKey{Name: ext.Name}})
```
nit: just to avoid the impression that ExtensionConfig is namespaced